src/main/PileupCache.js

```diff
@@ -91,13 +97,17 @@ class PileupCache {
     if (group.alignments.length == 1) {
       // This is the first read in the group. Infer its span from its mate properties.
       // TODO: if the mate Alignment is also available, it would be better to use that.
-      var intervals = [range];
-      var mateProps = read.getMateProperties();
-      if (mateProps && mateProps.ref && mateProps.ref == read.ref) {
-        mateInterval = new ContigInterval(mateProps.ref, mateProps.pos, mateProps.pos + range.length());
-        intervals.push(mateInterval);
+      if (this.viewAsPairs) {
+        var mateProps = read.getMateProperties();
+        var intervals = [range];
+        if (mateProps && mateProps.ref && mateProps.ref == read.ref) {
```

Comment:
Not at all necessary, just a thought; I've enjoyed using functions like `equalsIn(obj, path, val)` to handle this (in this case, `equalsIn(matProps, ['ref'], read.ref)`).